editor/src/components/editor/store/store-hook.ts

```diff
@@ -1,12 +1,30 @@
 import React from 'react'
-import type { EditorStorePatched } from './editor-state'
+import type { EditorState, EditorStoreFull, EditorStorePatched } from './editor-state'
 import { UseStore, StoreApi, EqualityChecker } from 'zustand'
 import { shallowEqual } from '../../../core/shared/equality-utils'
 import { isFeatureEnabled } from '../../../utils/feature-switches'
 import { PERFORMANCE_MARKS_ALLOWED } from '../../../common/env-vars'
 
 type StateSelector<T, U> = (state: T) => U
 
+export type PatchedSelector<U> = StateSelector<EditorStorePatched, U>
+
+export type FullSelector<U> = StateSelector<EditorStoreFull, U>
+
+export function patchedStoreFromFullStore(store: EditorStoreFull): EditorStorePatched {
```

Comment:
Yeah, agreed. I'm pretty sure this will end up with a lot more objects being created.

Reply:
> since we have a _lot_ of useEditorState hooks, I wonder if it would be measurably faster if we could skip making the `fullSelector` and also skip creating this intermediary object every time useEditorState is called.

I think the key bit here is "measurably faster", it's not showing up in the performance regression graphs at the very least. If we're really concerned about that I'd argue for memoising the value so that we only create the one per change to `EditorState`.

> For example if the context would already have the patched editor named as just `editor` then we could just omit unpatchedEditor from the type signature of `useEditorState`, not having to worry about it being "physically" there on the object

I'd argue that's a really bad idea because it is still there but we wouldn't know without debugging it which could easily lead to some weird bugs.